easybuild/easyconfigs/i/impi/impi-2019.8.254-iccifort-2020.3.275.eb

```
@@ -0,0 +1,40 @@
+# This is an easyconfig file for EasyBuild, see http://easybuilders.github.io/easybuild
+
+name = 'impi'
+version = '2019.8.254'
+
+homepage = 'https://software.intel.com/en-us/intel-mpi-library/'
+description = "Intel MPI Library, compatible with MPICH ABI"
+
+toolchain = {'name': 'iccifort', 'version': '2020.3.275'}
+
+source_urls = ['https://registrationcenter-download.intel.com/akdlm/irc_nas/tec/16814/']
+sources = ['l_mpi_%(version)s.tgz']
+checksums = ['fa163b4b79bd1b7509980c3e7ad81b354fc281a92f9cf2469bf4d323899567c0']
+
+dependencies = [
+    # needed by libfabric provider MLX introduced in Intel MPI v2019.6,
+    # https://software.intel.com/en-us/articles/improve-performance-and-stability-with-intel-mpi-library-on-infiniband
+    ('UCX', '1.9.0'),
```

It seems that they have disabled the `mlx` provider. The library is still there in `$EBROOTIMPI/intel64/libfabric/lib/prov/libmlx-fi.s` but it is not being used. Our machines with `ConnectX-4` and `ConnectX-5` pick `verbs` as the provider. It might be still be the case that `mlx` is now only used with `ConnectX-6`. I doubt it but I lack the hardware to test it. 

```suggestion
    # needed by libfabric provider MLX introduced in Intel MPI v2019.6,
    # https://software.intel.com/en-us/articles/improve-performance-and-stability-with-intel-mpi-library-on-infiniband
    # ('UCX', '1.9.0'),
```

---

@boegel @bartoldeman do you have ConnectX-6 to test which provider is used?

---

OK, here we go again, `fi_info` does *not* report the `mlx` provider as available, but it is still being used in our machines with Mellanox NCAs. Executing `mpirun` with `FI_LOG_LEVEL=debug` shows the following

```
$ FI_LOG_LEVEL=debug mpirun ./test
[...]
libfabric:131876:core:core:ofi_register_provider():418<info> registering provider: verbs (110.10)
libfabric:131876:core:core:ofi_register_provider():418<info> registering provider: mlx (1.4)
libfabric:131876:core:core:ofi_register_provider():418<info> registering provider: ofi_hook_noop (110.10)
libfabric:131876:core:core:fi_getinfo_():1066<info> Found provider with the highest priority mlx, must_use_util_prov = 0
libfabric:131876:mlx:core:mlx_getinfo():169<info> used inject size = 1024 
libfabric:131876:mlx:core:mlx_getinfo():204<info> Loaded MLX version 1.4.0
libfabric:131876:mlx:core:mlx_getinfo():242<warn> MLX: spawn support 0 
libfabric:131876:core:core:fi_getinfo_():1093<info> Since mlx can be used, verbs has been skipped. To use verbs, please, set FI_PROVIDER=verbs
```

So, UCX is still a dependency.